### ISSUE TITLE
MySQL Logger for Unlock Events

### ISSUE THREAD

**Issue by asmartin** [pgina/pgina#271]
Hello,

I am using pGina with the following plugins on Windows 7:

-    LDAP: authenticate users with an LDAP server
-    Local Machine - always authenticate local users
-    MySQL Logger - log all logon, logoff, lock, unlock, etc events to a database
-    Single User Login - login to Windows as a specific local user

This works very well whenever a new user logs on (the LDAP username is logged successfully in the MySQL database). However, if a user just locks the workstation and then someone else authenticates and unlocks it, the original username is logged to the MySQL database. Is it possible to have the username that actually unlocked the machine be logged to the MySQL database?

Thanks!

**Comment by nate-yocom** [comment]
I'm not sure I understand.  If user A locks the machine, only user A can unlock it.  How have you facilitated another user doing the unlock? 

Do you mean that the unlock is recorded using the Single User Login username, as opposed to the user's actual username as entered?

**Comment by asmartin** [comment]
Correct, the Single User Login username is what gets recorded by MySQL Logger, but what I would like is that the username of the actual LDAP user who logged in gets logged to MySQL instead. Is this possible?

**Comment by nate-yocom** [comment]
I believe so.  We'll keep it open as a bug for now and advise.

**Comment by nate-yocom** [comment]
My reading of the plugin is that it should be using the correct username.  What version of pGina are you using? and do you have the MySQL Logger setting "UseModifiedName" set in the registry to anything?

**Comment by asmartin** [comment]
I'm using pGina 3.1.8.0 with `UseModifiedName` set to `False` in the registry

**Comment by nate-yocom** [comment]
Can you validate whether the beta version of pGina resolves this?
